File: solutions/python/bob/1/bob.py
def response(hey_bob: str):
    hey_bob = hey_bob.strip()

    if not hey_bob:
        return "Fine. Be that way!"

    is_question = hey_bob.endswith("?")

    is_yelling = hey_bob.isupper()

    if is_yelling and is_question:
        return "Calm down, I know what I'm doing!"
    elif is_yelling:
        return "Whoa, chill out!"
    elif is_question:
        return "Sure."
    else:
        return "Whatever."

File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
    const msg = message.trim()
    if(!msg) return "Fine. Be that way!";

    let is_question = msg.endsWith('?');
    let has_letters = [...msg].some(is_alphabetic);
    let is_yelling = has_letters &&  msg === msg.toLocaleUpperCase();

    switch (true) {
        case is_yelling && is_question:
          return "Calm down, I know what I'm doing!";
        case is_yelling:
          return "Whoa, chill out!";
        case is_question:
          return "Sure.";
        default:
          return "Whatever.";
      }

}

const is_alphabetic = (char: string) => /^\p{L}$/u.test(char);
